core/src/Kokkos_Core_fwd.hpp

```diff
@@ -220,13 +226,13 @@ class LogicalMemorySpace;
 
 }  // namespace Kokkos
 
-#define KOKKOS_RESTRICT_EXECUTION_TO_DATA(DATA_SPACE, DATA_PTR) \
-  Kokkos::Impl::VerifyExecutionCanAccessMemorySpace<            \
-      Kokkos::Impl::ActiveExecutionMemorySpace, DATA_SPACE>::verify(DATA_PTR)
+#define KOKKOS_RESTRICT_EXECUTION_TO_DATA(DATA_SPACE, DATA_PTR)        \
+  Kokkos::Impl::verify_space<Kokkos::Impl::ActiveExecutionMemorySpace, \
+                             DATA_SPACE>::check();
```

Comment:
Non-blocking: Why is this still a macro? Seems detrimental to maintainability, but I understand if you want to minimize the size of the diff.

Reply:
This macro is used outside of `Kokkos`. Otherwise, we would just remove it.